ktor-network/common/test/io/ktor/network/sockets/tests/TCPSocketTest.kt

```diff
@@ -175,13 +254,56 @@ class TCPSocketTest {
             .tcp()
             .bind(InetSocketAddress("127.0.0.1", 0))
 
-        launch {
-            assertFailsWith<IOException> {
+        val acceptJob = launch {
+            // The accept call should fail with IOException/PosixException because the socket was closed,
+            // but it must not be a bad descriptor error caused by closed descriptor in select call.
+            try {
                 socket.accept()
+            } catch (exception: IOException) {
+                assertFalse("Bad descriptor" in exception.message.orEmpty())
+            } catch (exception: Exception) {
+                assertTrue(exception.isPosixException())
             }
```

Comment:
I don't like how this is done, it would be better if `PosixException` extends `IOException`. That would make catching this exception in common code easier. Why is that not already the case?